js/blocks/controls/user-input.js

```diff
@@ -0,0 +1,298 @@
+			// In the Windows version of Firefox the up and down arrows don't move the caret
+			// within an input field like they do for Mac Firefox/Chrome/Safari. This causes
+			// a form of focus trapping that is disruptive to the user experience. This disruption
+			// only happens if the caret is not in the first or last position in the text input.
+			// See: https://github.com/WordPress/gutenberg/issues/5693#issuecomment-436684747
+			switch ( event.keyCode ) {
+				// When UP is pressed, if the caret is at the start of the text, move it to the 0
+				// position.
+				case UP: {
+					if ( 0 !== event.target.selectionStart ) {
+						event.stopPropagation();
+						event.preventDefault();
+
+						// Set the input caret to position 0
+						event.target.setSelectionRange( 0, 0 );
+					}
+					break;
+				}
+				// When DOWN is pressed, if the caret is not at the end of the text, move it to the
+				// last position.
+				case DOWN: {
+					if ( this.props.value.length !== event.target.selectionStart ) {
+						event.stopPropagation();
+						event.preventDefault();
+
+						// Set the input caret to the last position
+						event.target.setSelectionRange( this.props.value.length, this.props.value.length );
+					}
+					break;
+				}
+			}
+
+			return;
+		}
+
+		const user = this.state.users[ this.state.selectedSuggestion ];
+
+		switch ( event.keyCode ) {
+			case UP: {
+				event.stopPropagation();
+				event.preventDefault();
+				const previousIndex = ! selectedSuggestion ? users.length - 1 : selectedSuggestion - 1;
+				this.setState( {
+					selectedSuggestion: previousIndex,
+				} );
+				break;
+			}
+			case DOWN: {
+				event.stopPropagation();
+				event.preventDefault();
+				const nextIndex = selectedSuggestion === null || ( selectedSuggestion === users.length - 1 ) ? 0 : selectedSuggestion + 1;
+				this.setState( {
+					selectedSuggestion: nextIndex,
+				} );
+				break;
+			}
+			case TAB: {
+				if ( this.state.selectedSuggestion !== null ) {
+					this.selectLink( user );
+					// Announce a link has been selected when tabbing away from the input field.
+					this.props.speak( __( 'User selected.', 'block-lab' ) );
+				}
+				break;
+			}
+			case ENTER: {
+				if ( this.state.selectedSuggestion !== null ) {
+					event.stopPropagation();
+					this.selectLink( user );
+				}
+				break;
+			}
+		}
+	}
+
+	selectLink( user ) {
+		this.props.onChange( user.slug );
+		this.setState( {
+			selectedSuggestion: null,
+			showSuggestions: false,
+		} );
+	}
+
+	handleOnClick( user ) {
+		this.selectLink( user );
+		// Move focus to the input field when a link suggestion is clicked.
+		this.inputRef.current.focus();
+	}
+
+	render() {
+		const { value = '', autoFocus = true, instanceId, className, placeholder } = this.props;
+		const { showSuggestions, users, selectedSuggestion, loading } = this.state;
+		/* eslint-disable jsx-a11y/no-autofocus */
+		return (
+			<div className={ classnames( 'editor-url-input', className ) }>
```

Comment:
This seems to be missing the field's label? For other controls, I've used a `<BaseControl>` instead of a `<div>` here, like so:
```
<BaseControl label={field.label} className="{ classnames( 'editor-url-input', className ) }" help={field.help}>
    …
</BaseControl>

Reply:
![screen shot 2019-03-03 at 7 33 26 pm](https://user-images.githubusercontent.com/1097667/53693379-54279580-3deb-11e9-8a9b-187c2eb2439d.png)

Reply:
Great point.

https://github.com/getblocklab/block-lab/pull/233/commits/d80465db25a7c78261801ab85f92e9675cf06b13 uses your suggestion of changing the `<div>` to a `<BaseControl>`. Now, the label displays:

<img width="980" alt="label-now-exists" src="https://user-images.githubusercontent.com/4063887/53705792-8980b300-3dec-11e9-8be7-c76919f333d4.png">